wasm/src/lib.rs

```diff
@@ -0,0 +1,148 @@
+extern crate ed25519_dalek;
+extern crate hex;
+
+mod utils;
+
+use wasm_bindgen::prelude::*;
+
+use std::convert::TryFrom;
+use ed25519_dalek::{Keypair, PublicKey, SecretKey, Signature, Signer as _, Verifier as _, SECRET_KEY_LENGTH, PUBLIC_KEY_LENGTH};
+use serde::{Deserialize, Serialize};
+use sha2::{Digest, Sha256};
+
+cfg_if::cfg_if! {
+	if #[cfg(consolelog)] {
+		use web_sys::console;
+
+		fn console_log(info: &str) {
+			console::log_1(&info.into());
+		}
+	} else {
+		fn console_log(_info: &str) {}
+	}
+}
+
+#[wasm_bindgen]
+pub fn setup() {
+	console_log("@solana/wasm Initialized. WASM ready for calls.");
+	utils::set_panic_hook();
+}
+
+#[wasm_bindgen]
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct Pair{
+	public_key: Vec<u8>,
+	secret_key: Vec<u8>
+}
+
+/// Generate key pair for use with ED25519 from ed25519_dalek
+///
+/// * returned struct has two fields public and secret.
+#[wasm_bindgen(js_name = generateKeyPair)]
+pub fn generate_keypair() -> Result<JsValue, JsValue> {
+	// using get random directly instead of OsRng since it's easier to integrate with WASM
+	let mut seed = [0u8; 32];
+	getrandom::getrandom(&mut seed).unwrap();
+	let secret = SecretKey::from_bytes(&seed).unwrap();
+	let public: PublicKey = (&secret).into();
+	let keypair = Keypair{secret, public};
+
+	// use all 64bit for secret key to maintain compatibility with tweetnacl
+	let secret_key = keypair.to_bytes().to_vec();
+	let public_key = keypair.public.as_bytes().to_vec();
+
+	let result = Pair{ public_key, secret_key};
+	Ok(JsValue::from_serde(&result).unwrap())
```

Comment:
You can return `Pair` directly, the serialization will be done automatically and then you will get nice typescript typedefs